test/operator.jl

```diff
@@ -247,4 +247,281 @@ using Compat.Test
+                                        x[1] + 2x[2] +  x[3] + 2
+                                        x[2] + 2x[3] + 2])
+            @test vec_eq(2ones(3) - y, [-2x[1] -  x[2]         + 2
+                                        -x[1] - 2x[2] -  x[3] + 2
+                                        -x[2] - 2x[3] + 2])
+            @test vec_eq(y + x, [3x[1] +  x[2]
+                                 x[1] + 3x[2] +  x[3]
+                                 x[2] + 3x[3]])
+            @test vec_eq(x + y, [3x[1] +  x[2]
+                                 x[1] + 3x[2] +  x[3]
+                                 x[2] + 3x[3]])
+            @test vec_eq(2y + 2x, [6x[1] + 2x[2]
+                                   2x[1] + 6x[2] + 2x[3]
+                                   2x[2] + 6x[3]])
+            @test vec_eq(y - x, [ x[1] + x[2]
+                                 x[1] + x[2] + x[3]
+                                 x[2] + x[3]])
+            @test vec_eq(x - y, [-x[1] - x[2]
+                                 -x[1] - x[2] - x[3]
+                                 -x[2] - x[3]])
+            @test vec_eq(y + x[:], [3x[1] +  x[2]
+                                    x[1] + 3x[2] +  x[3]
+                                    x[2] + 3x[3]])
+            @test vec_eq(x[:] + y, [3x[1] +  x[2]
+                                    x[1] + 3x[2] +  x[3]
+                                    x[2] + 3x[3]])
+
+            @test vec_eq(@JuMP.Expression(A*x/2), A*x/2)
+            @test vec_eq(X*v,  [4X11; 6X23; 0])
+            @test vec_eq(v'*X,  [4X11  0   5X23])
+            @test vec_eq(v.'*X, [4X11  0   5X23])
+            @test vec_eq(X'*v,  [4X11;  0;  5X23])
+            @test vec_eq(X.'*v, [4X11; 0;  5X23])
+            @test vec_eq(X*A,  [2X11  X11  0
+                                0     X23  2X23
+                                0     0    0   ])
+            @test vec_eq(A*X,  [2X11  0    X23
+                                X11   0    2X23
+                                0     0    X23])
+            @test vec_eq(A*X', [2X11  0    0
+                                X11   X23  0
+                                0     2X23 0])
+            @test vec_eq(X'*A, [2X11  X11  0
+                                0     0    0
+                                X23   2X23 X23])
+            @test vec_eq(X.'*A, [2X11 X11  0
+                                 0    0    0
+                                 X23  2X23 X23])
+            @test vec_eq(A'*X, [2X11  0 X23
+                                X11   0 2X23
+                                0     0 X23])
+            @test vec_eq(X.'*A, X'*A)
+            @test vec_eq(A.'*X, A'*X)
+            @test vec_eq(X*A, X*B)
+            @test vec_eq(Y'*A, Y.'*A)
+            @test vec_eq(A*Y', A*Y.')
+            @test vec_eq(Z'*A, Z.'*A)
+            @test vec_eq(Xd'*Y, Xd.'*Y)
+            @test vec_eq(Y'*Xd, Y.'*Xd)
+            @test vec_eq(Xd'*Xd, Xd.'*Xd)
+            # @test_broken vec_eq(A*X, B*X)
```

Comment:
Why is this code commented?

Reply:
It was commented out in `old/operators.jl`, I have uncommented them and checked what were the issue, see the description of this PR for details :)